src/modules/form/hooks/useFormDialog.tsx

```diff
@@ -1,49 +1,197 @@
-import { Dialog, DialogContent, DialogTitle } from '@mui/material';
-import { useCallback, useState } from 'react';
+import { LoadingButton } from '@mui/lab';
+import {
+  Box,
+  Button,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Grid,
+  Stack,
+} from '@mui/material';
+import { ReactNode, useCallback, useRef, useState } from 'react';
 
-import { SubmitFormAllowedTypes } from '../types';
+import DynamicForm, {
+  DynamicFormOnSubmit,
+  DynamicFormProps,
+  DynamicFormRef,
+} from '../components/DynamicForm';
+import { LocalConstants, SubmitFormAllowedTypes } from '../types';
+import {
+  createHudValuesForSubmit,
+  createValuesForSubmit,
+} from '../util/formUtil';
 
-import EditRecord, {
-  Props as EditRecordProps,
-} from '@/modules/form/components/EditRecord';
+import useFormDefinition from './useFormDefinition';
+import useInitialFormValues from './useInitialFormValues';
+
+import CommonDialog from '@/components/elements/CommonDialog';
+import Loading from '@/components/elements/Loading';
+import NotFound from '@/components/pages/NotFound';
+import {
+  emptyErrorState,
+  ErrorState,
+  partitionValidations,
+} from '@/modules/errors/util';
+import { FormInput, FormRole, useSubmitFormMutation } from '@/types/gqlTypes';
 import { PartialPick } from '@/utils/typeUtil';
 
-export function useFormDialog<T extends SubmitFormAllowedTypes>() {
+interface Args<T> {
+  formRole: FormRole;
+  record?: T;
+  onCompleted?: (data: T) => void;
+  localConstants?: LocalConstants;
+  inputVariables?: Omit<
+    FormInput,
+    'confirmed' | 'formDefinitionId' | 'values' | 'hudValues' | 'recordId'
+  >;
+}
+
+type RenderFormDialogProps = PartialPick<
+  DynamicFormProps,
+  'onSubmit' | 'definition' | 'errors'
+> & {
+  title: ReactNode;
+};
+
+export function useFormDialog<T extends SubmitFormAllowedTypes>({
+  formRole,
+  record,
+  onCompleted,
+  localConstants,
+  inputVariables,
+}: Args<T>) {
   const [dialogOpen, setDialogOpen] = useState<boolean>(false);
   const openFormDialog = useCallback(() => setDialogOpen(true), []);
-  const renderFormDialog = ({
-    title,
-    ...props
-  }: PartialPick<EditRecordProps<T>, 'onCompleted' | 'title'>) => {
+  const {
+    formDefinition,
+    itemMap,
+    loading: definitionLoading,
+  } = useFormDefinition(formRole);
+  const formRef = useRef<DynamicFormRef>(null);
+  const [errors, setErrors] = useState<ErrorState>(emptyErrorState);
+  const [submitLoading, setSubmitLoading] = useState(false);
+  const closeDialog = useCallback(() => {
+    setDialogOpen(false);
+    setErrors(emptyErrorState);
+    setSubmitLoading(false);
+  }, []);
+
+  const [submitForm, { loading: saveLoading }] = useSubmitFormMutation({
```

Comment:
This is a lot of repeated code. I'm working on a separate branch to pull it into a hook, since its now used in several places.